domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -1295,76 +1266,141 @@ public void update(String table, Object entity, String id, Handler<AsyncResult<R
    */
+  public Future<RowSet<Row>> update(String table, Object entity, Criterion filter, boolean returnUpdatedIds) {
+    return withConn(conn -> conn.update(table, entity, filter, returnUpdatedIds));
+  }
+
+  /**
+   * Update 1...n records matching the filter
+   * <br>
+   * Criterion Examples:
+   * <br>
+   * 1. can be mapped from a string in the following format [{"field":"''","value":"","op":""}]
+   * <pre>
+   *    Criterion a = json2Criterion("[{\"field\":\"'fund_distributions'->[]->'amount'->>'sum'\",\"value\":120,\"op\":\"<\"}]"); //denotes funds_distribution is an array of objects
+   *    Criterion a = json2Criterion("[{"field":"'po_line_status'->>'value'","value":"SENT","op":"like"},{"field":"'owner'->>'value'","value":"MITLIBMATH","op":"="}, {"op":"AND"}]");
+   *    (see postgres query syntax for more examples in the read.me
+   * </pre>
+   * 2. Simple Criterion
+   * <pre>
+   *    Criteria b = new Criteria();
+   *    b.field.add("'note'");
+   *    b.operation = "=";
+   *    b.value = "a";
+   *    b.isArray = true; //denotes that the queried field is an array with multiple values
+   *    Criterion a = new Criterion(b);
+   * </pre>
+   * 3. For a boolean field called rush = false OR note[] contains 'a'
+   * <pre>
+   *    Criteria d = new Criteria();
+   *    d.field.add("'rush'");
+   *    d.operation = Criteria.OP_IS_FALSE;
+   *    d.value = null;
+   *    Criterion a = new Criterion();
+   *    a.addCriterion(d, Criteria.OP_OR, b);
+   * </pre>
+   * 4. for the following json:
+   * <pre>
+   *      "price": {
+   *        "sum": "150.0",
+   *         "po_currency": {
+   *           "value": "USD",
+   *           "desc": "US Dollar"
+   *         }
+   *       },
+   *
+   *    Criteria c = new Criteria();
+   *    c.addField("'price'").addField("'po_currency'").addField("'value'");
+   *    c.operation = Criteria.OP_LIKE;
+   *    c.value = "USD";
+   *
+   * </pre>
+   * @param table - table to update
+   * @param entity - pojo to set for matching records
+   * @param filter - see example below
+   * @param returnUpdatedIds - return ids of updated records
+   */
   public void update(String table, Object entity, Criterion filter, boolean returnUpdatedIds,
-                     Handler<AsyncResult<RowSet<Row>>> replyHandler)
-  {
-    String where = null;
-    if(filter != null){
-      where = filter.toString();
-    }
-    update(table, entity, DEFAULT_JSONB_FIELD_NAME, where, returnUpdatedIds, replyHandler);
+                     Handler<AsyncResult<RowSet<Row>>> replyHandler) {
+    update(table, entity, filter, returnUpdatedIds).onComplete(replyHandler);
+  }
+
+  /**
+   * Update all records in {@code table} that match the {@code CQLWrapper} query.
+   * @param entity new content for the matched records
+   * @return one row with the id for each updated record if returnUpdatedIds is true
+   */
+  public Future<RowSet<Row>> update(String table, Object entity, CQLWrapper filter, boolean returnUpdatedIds) {
```

Comment:
Fix Javadoc

Reply:
`@param` javadoc is not needed if the param is sufficiently explained in the summary: https://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment